### PR TITLE
docs(ch62): coding standards

### PR DESCRIPTION
**Pull request by ryexley**
[CH62](https://app.clubhouse.io/aoeu-se/story/62/document-coding-standards-for-javascript)

This PR adds extensive documentation on our JavaScript and React/JSX coding standards, and specifically our custom ESLint configuration, as well as some baseline information for our Python and Go coding standards as well.

[Preview can be reviewed on Vercel here](https://engineering-docs-7y0hdhk16-aoeu-se.vercel.app/pages/coding-standards/javascript-react).

### CODE REVIEW

File: docs/coding-standards/javascript-react.mdx [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+
+ESLint has the benefit of providing functionality to both check and enforce our coding standards, but also perform static analysis for certain elements of code quality as well, such as [cyclomatic complexity](https://en.wikipedia.org/wiki/Cyclomatic_complexity). This helps us keep our code quality high, but has the additional benefit of helping us focus the tests we write for our code as well.
+
+If you wish to see the details of our JavaScript and React coding standards, you are encouraged to review that/those configurations and/or documentation.
+
+<!--
```
Comment (theantichris) [inline docs/coding-standards/javascript-react.mdx:14]: 😏

**Review by theantichris — APPROVED** [review]

File: docs/coding-standards/javascript-react.mdx (hunk 1 above)
Comment (ryexley) [reply docs/coding-standards/javascript-react.mdx:14]: Just like a boy scout ... _always ready_ 😉

**Review by theantichris — APPROVED** [review]

File: docs/coding-standards/javascript-react.mdx [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+Bad:
+
+```
+const foo = "bar";
+```
+
+Good:
+
+```
+const foo = bar
```
Comment (jmosley5) [inline docs/coding-standards/javascript-react.mdx:136]: Does 'bar' need the "quotes" around it? It looks like _Good_ = without the semi-colon and the quotes.

**Review by jmosley5 — CHANGES_REQUESTED** [review]

File: docs/coding-standards/javascript-react.mdx (hunk 2 above)
Comment (theantichris) [reply docs/coding-standards/javascript-react.mdx:136]: I think in this case `bar` is another variable name so wouldn't need the quotes.

Comment (ryexley) [reply docs/coding-standards/javascript-react.mdx]: ✅

**Review by jmosley5 — APPROVED** [review]

File: docs/coding-standards/javascript-react.mdx [hunk 3]
```diff
@@ -133,7 +133,7 @@ const foo = "bar";
 Good:
 
 ```
-const foo = bar
+const foo = "bar"
```
Comment (theantichris) [inline docs/coding-standards/javascript-react.mdx:136]: I guess I was wrong @jmosley5 😂

**Review by theantichris — APPROVED** [review]

**Pull request merged**